gcc/brig/brig-lang.c

```diff
@@ -128,6 +128,11 @@ brig_langhook_init_options_struct (struct gcc_options *opts)
 
   opts->x_flag_finite_math_only = 0;
   opts->x_flag_signed_zeros = 1;
+
+  /* Prevent some optimizations that break HSAIL's ftz semantics. For
+     example, X * (+/-)1.0 -> (+/-)X simplification is not valid for
+     denormals. */
+  opts->x_flag_ftz_math = 1;
```

Comment:
Please create a separate pull request (or at least a commit) which has only the -fftz-math core updates as we need to ask for a wider review for it before committing it to gcc trunk. The flag would assume the runtime switches CPU to a FTZ/DAZ mode or has it on by default and affects only compile-time assumptions of values. That is, it would not inject any runtime calls to toggle the flags (that's gccbrig's job).